src/containers/RecentTrades/Market.tsx

```diff
@@ -49,91 +30,75 @@ const handleHighlightValue = (prevValue: string, curValue: string) => {
     );
 };
 
+const RecentTradesMarket = () => {
+    const { formatMessage } = useIntl();
+    const dispatch = useDispatch();
 
-class RecentTradesMarketContainer extends React.Component<Props> {
-    public componentDidMount() {
-        if (this.props.currentMarket) {
-            this.props.tradesFetch(this.props.currentMarket);
-        }
-    }
-
-    public componentWillReceiveProps(next: Props) {
-        if (next.currentMarket && this.props.currentMarket !== next.currentMarket) {
-            this.props.tradesFetch(next.currentMarket);
-        }
-    }
-
-    public shouldComponentUpdate(nextProps: Props) {
-        return JSON.stringify(nextProps.recentTrades) !== JSON.stringify(this.props.recentTrades);
-    }
+    const currentMarket = useSelector(selectCurrentMarket);
+    const currentPrice = useSelector(selectCurrentPrice);
+    const recentTrades = useSelector(selectRecentTradesOfCurrentMarket);
 
-    public render() {
-        return (
-            <div className="pg-recent-trades__markets">
-                <Table
-                    data={this.getTrades(this.props.recentTrades)}
-                    header={this.getHeaders()}
-                    onSelect={this.handleOnSelect}
-                />
-            </div>
-        );
-    }
-
-    private getHeaders = () => {
+    const headers = React.useMemo(() => {
         return [
-            this.props.intl.formatMessage({ id: 'page.body.trade.header.recentTrades.content.time' }),
-            this.props.intl.formatMessage({ id: 'page.body.trade.header.recentTrades.content.amount' }),
-            this.props.intl.formatMessage({ id: 'page.body.trade.header.recentTrades.content.price' }),
+            formatMessage({ id: 'page.body.trade.header.recentTrades.content.time' }),
+            formatMessage({ id: 'page.body.trade.header.recentTrades.content.amount' }),
+            formatMessage({ id: 'page.body.trade.header.recentTrades.content.price' }),
         ];
-    };
+    }, [formatMessage]);
 
-    private getTrades(trades: PublicTrade[]) {
-        const priceFixed = this.props.currentMarket ? this.props.currentMarket.price_precision : 0;
-        const amountFixed = this.props.currentMarket ? this.props.currentMarket.amount_precision : 0;
+    const getTrades = () => {
```

Comment:
use `useCallback` hook for ALL inner methods inside component

Reply:
`getTrades` has a dependency `recentTrades` which is an array, so If we include it in useCallback's dependencies it will be still re-creating a func every time. If I am not mistaken it will be the same as with useCallback
Won't it?

Reply:
@berezh

Reply:
does not matter you think it will be the same with useCallback or without, use useCallback in all cases to have one standard.

Reply:
ok, deal